web-interface/src/components/system/authentication/MFASetupStep1.jsx

```diff
@@ -21,9 +21,35 @@ function MFASetupStep1(props) {
 
         <p className="mt-4">
           Scan the QR code with your preferred TOTP application like Google Authenticator, Authy
-          or the Yubikey Authenticator.
+          or the Yubikey Authenticator.{' '}
+          <a href="#" data-bs-toggle="modal" data-bs-target="#otp-url-modal">Show TOTP URL</a>
         </p>
 
+        <div className="modal fade"
+             id="otp-url-modal"
+             tabIndex="-1"
+             aria-labelledby="otp-url-modal-label"
+             aria-hidden="true">
+          <div className="modal-dialog">
+            <div className="modal-content">
+              <div className="modal-body">
+                <h2>TOTP URL</h2>
+
+                <p>
+                  Many TOTP applications allow you to enter the TOTP URL directly f you cannot scan the QR code:
```

Comment:
Typo on "f you cannot".
```suggestion
                  Many TOTP applications allow you to enter the TOTP URL directly if you cannot scan the QR code:
```